optics-core/src/Optics/Monoidal.hs

```diff
@@ -0,0 +1,193 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE UndecidableInstances #-}
+module Optics.Monoidal where
+
+import GHC.TypeLits
+import Optics.Fold
+import Optics.IxFold
+import Optics.IxSetter
+import Optics.IxTraversal
+import Optics.Internal.Optic.Types
+import Optics.Optic
+import Optics.Setter
+import Optics.Traversal
+
+-- Essentially this collapses the hierarchy into
+-- Traversal, Fold, and Setter,
+-- and then takes the usual Join
+type family AfterJoin (k :: OpticKind) (l :: OpticKind) where
```

Comment:
Perhaps I'm missing something, but is it the case that `AfterJoin k l ~ Join A_Traversal (Join k l)`? Couldn't we use that definition instead?

Reply:
I haven't thought deeply about it, but I guess that would make the trick, since `A_Traversal` is the "bottom" of that restricted hierarchy.